bitsandbytes/cuda_setup/main.py

```diff
@@ -37,7 +37,7 @@
     # we have libcudart.so.11.0 which causes a lot of errors before
     # not sure if libcudart.so.12.0 exists in pytorch installs, but it does not hurt
     CUDA_RUNTIME_LIBS = ["libcudart.so", "libcudart.so.11.0", "libcudart.so.12.0", "libcudart.so.12.1", "libcudart.so.12.2"]
-    DYNAMIC_LIBRARY_SUFFIX = ".so"
+    DYNAMIC_LIBRARY_SUFFIX = { "Darwin": ".dylib", "Windows":".dll" }.get(platform.system(), ".so")
```

Comment:
Using a dict for this is a great idea, but since we check for `if platform.system() == 'Windows'` above, this should be simply moved out of the `else` branch (and the `DYNAMIC_LIBRARY_SUFFIX = ".dll"` above in the Windows branch then eliminated).